src/REVLoans.sol

```diff
@@ -1102,19 +1114,21 @@ contract REVLoans is ERC721, ERC2771Context, IREVLoans, Ownable {
     /// @param loan The loan being paid off.
     /// @param revnetId The ID of the revnet the loan is being paid off in.
     /// @param amount The amount being paid off.
-    /// @param sourceFeeAmount The amount of the fee being taken from the revnet acting as the source of the loan.
-    function _removeFrom(REVLoan memory loan, uint256 revnetId, uint256 amount, uint256 sourceFeeAmount) internal {
+    function _removeFrom(REVLoan memory loan, uint256 revnetId, uint256 amount) internal {
         // Decrement the total amount of a token being loaned out by the revnet from its terminal.
         totalBorrowedFrom[revnetId][loan.source.terminal][loan.source.token] -= amount;
 
+        // Increase the allowance for the beneficiary.
+        _beforeTransferTo({to: address(loan.source.terminal), token: loan.source.token, amount: amount});
+
         // The borrowed amount to return to the revnet.
         uint256 payValue = loan.source.token == JBConstants.NATIVE_TOKEN ? amount : 0;
 
         // Add the loaned amount back to the revnet.
         try loan.source.terminal.addToBalanceOf{value: payValue}({
             projectId: revnetId,
             token: loan.source.token,
-            amount: _balanceOf(loan.source.token) - sourceFeeAmount,
```

Comment:
Shouldn't we still leave `sourceFeeAmount` in the contract to be paid to the source fee project?

Reply:
we should. theoretically `amount` should already take this into account. right now, the `payValue` is calculated using `amount`, but the `amount` param in the `addToBalanceOf` fn is the result of this subtraction. I figured they both evaluate to the same thing, and we should just use one of them. thoughts?